qa/tasks/cephfs/xfstests_dev.py

```diff
@@ -110,11 +110,13 @@ def install_deps(self):
         distro = distro.lower()
         major_ver_num = int(version.split('.')[0]) # only keep major release
                                                    # number
+        logger.info(f'distro detected is - {distro}. major version number '
+                    f'is - {major_ver_num}')
 
         # we keep fedora here so that right deps are installed when this test
         # is run locally by a dev.
         if distro in ('redhatenterpriseserver', 'redhatenterprise', 'fedora',
-                      'centos', 'centosstream'):
+                      'centos', 'centosstream', 'rhel'):
```

Comment:
Okay. As I remembered it will be any of `redhatenterpriseserver` or `redhatenterprise`:

Checked the logs it's really `rhel` for this case:
```

2023-02-14T03:58:35.837 INFO:teuthology.orchestra.run.smithi105.stdout:x86_64
2023-02-14T03:58:35.840 DEBUG:teuthology.orchestra.run.smithi105:> cat /etc/os-release
2023-02-14T03:58:35.893 INFO:teuthology.orchestra.run.smithi105.stdout:NAME="Red Hat Enterprise Linux"
2023-02-14T03:58:35.893 INFO:teuthology.orchestra.run.smithi105.stdout:VERSION="8.4 (Ootpa)"
2023-02-14T03:58:35.894 INFO:teuthology.orchestra.run.smithi105.stdout:ID="rhel"
2023-02-14T03:58:35.894 INFO:teuthology.orchestra.run.smithi105.stdout:ID_LIKE="fedora"
2023-02-14T03:58:35.894 INFO:teuthology.orchestra.run.smithi105.stdout:VERSION_ID="8.4"
2023-02-14T03:58:35.894 INFO:teuthology.orchestra.run.smithi105.stdout:PLATFORM_ID="platform:el8"
2023-02-14T03:58:35.894 INFO:teuthology.orchestra.run.smithi105.stdout:PRETTY_NAME="Red Hat Enterprise Linux 8.4 (Ootpa)"
2023-02-14T03:58:35.894 INFO:teuthology.orchestra.run.smithi105.stdout:ANSI_COLOR="0;31"
2023-02-14T03:58:35.894 INFO:teuthology.orchestra.run.smithi105.stdout:CPE_NAME="cpe:/o:redhat:enterprise_linux:8.4:GA"
2023-02-14T03:58:35.895 INFO:teuthology.orchestra.run.smithi105.stdout:HOME_URL="https://www.redhat.com/"
2023-02-14T03:58:35.895 INFO:teuthology.orchestra.run.smithi105.stdout:DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/8/"
2023-02-14T03:58:35.895 INFO:teuthology.orchestra.run.smithi105.stdout:BUG_REPORT_URL="https://bugzilla.redhat.com/"
2023-02-14T03:58:35.895 INFO:teuthology.orchestra.run.smithi105.stdout:
2023-02-14T03:58:35.895 INFO:teuthology.orchestra.run.smithi105.stdout:REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
2023-02-14T03:58:35.895 INFO:teuthology.orchestra.run.smithi105.stdout:REDHAT_BUGZILLA_PRODUCT_VERSION=8.4
2023-02-14T03:58:35.895 INFO:teuthology.orchestra.run.smithi105.stdout:REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
2023-02-14T03:58:35.896 INFO:teuthology.orchestra.run.smithi105.stdout:REDHAT_SUPPORT_PRODUCT_VERSION="8.4"

```

Reply:
Yes, it's really 'rhel' for this case which is happening for first time.

Reply:
Wait,  the `lsb-release -is` will read the conents from `/etc/redhat-release` instead.

Locally for my VM running rhel:

```
  1 NAME="Red Hat Enterprise Linux"                                                                 
  2 VERSION="8.6 (Ootpa)"
  3 ID="rhel"
  4 ID_LIKE="fedora"
  5 VERSION_ID="8.6"
  6 PLATFORM_ID="platform:el8"
  7 PRETTY_NAME="Red Hat Enterprise Linux 8.6 (Ootpa)"
  8 ANSI_COLOR="0;31"
  9 CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
 10 HOME_URL="https://www.redhat.com/"
 11 DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/8/"
 12 BUG_REPORT_URL="https://bugzilla.redhat.com/"
 13 
 14 REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
 15 REDHAT_BUGZILLA_PRODUCT_VERSION=8.6
 16 REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
 17 REDHAT_SUPPORT_PRODUCT_VERSION="8.6"

```

It will return:

```
# lsb_release -is
RedHatEnterprise
# cat /etc/redhat-release 
Red Hat Enterprise Linux release 8.6 (Ootpa)
```

I changed the `/etc/redhat-release` and it will be:
```
# cat /etc/redhat-release 
Red123 Hat Enterprise Linux release 8.6 (Ootpa)
# lsb_release -is
Red123HatEnterprise
```

Reply:
Okay. But xfstests_dev.py doesn't the change contents of `/etc/redhat-release`. It receives `distro` and `version` from `teuthology.misc.get_system_type()` which in turn runs `lsb_release` command.

Reply:
Yeah. You can just run `cat /etc/system-release` somewhere to debug and confirm it for the next run of the test.

Reply:
Okay. I'll add that change and re-run.

Reply:
BTW, why do we need this to check /etc/system-release? If test_acls.py runs fine with this patch, this PR is perfect, right?

Reply:
Yeah, from the `/etc/system-release` we can know what exactly it should be also.

Reply:
From `/a/rishabh-2023-02-17_06:40:41-fs-wip-yuri5-testing-2023-02-07-0850-quincy-distro-default-smithi/7178801/teuthology.log` -

```
2023-02-17T06:59:20.793 INFO:tasks.cephfs.xfstests_dev:distro detected is "rhel". version detected is - "8.6"
2023-02-17T06:59:20.793 DEBUG:teuthology.orchestra.run.smithi097:> cat /etc/system-release
2023-02-17T06:59:20.849 INFO:teuthology.orchestra.run.smithi097.stdout:Red Hat Enterprise Linux release 8.6 (Ootpa)
2023-02-17T06:59:20.849 INFO:tasks.cephfs.xfstests_dev:Red Hat Enterprise Linux release 8.6 (Ootpa)
```

Reply:
Okay, a little odd. Why different version return different values ?

Reply:
> Okay. But xfstests_dev.py doesn't the change contents of `/etc/redhat-release`. It receives `distro` and `version` from `teuthology.misc.get_system_type()` which in turn runs `lsb_release` command.

That's odd. If that's the case then get_system_type() needs to include "rhel" in this check:

```
    if system_value in ['CentOS', 'Fedora', 'RedHatEnterpriseServer',
                        'RedHatEnterprise',
                        'CentOSStream',
                        'openSUSE', 'openSUSE project', 'SUSE', 'SUSE LINUX']:
        return "rpm"
```

Reply:
It's happening only recently.